core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/wrapper/shape/BackwardChainingShapeSource.java

```diff
@@ -81,9 +81,6 @@ public Stream<ShapesGraph> getAllShapeContexts() {
 
 			List<? extends Statement> collect = stream.collect(Collectors.toList());
 
-			if (collect.size() > 0) {
```

Comment:
Does this belong is this PR? I prefer self-contained PRs.

Reply:
It should be in it's own PR.